src/components/main/filmsList/FilmsList.tsx

```diff
@@ -1,63 +1,98 @@
 import { useState, useEffect } from "react"
+import { ActionTypeInterface } from "../../../global"
 import FilmItem from "./filmItem/FilmItem"
 import { useSelector, useDispatch } from "react-redux"
-import { changeFilmCategoriesAndCount } from "../../../redux/filmList/action"
+import { changeFilmCategoriesAndCount } from "../../../redux/filmListCountAndCategories/action"
 import {
     FilmInterface,
     getFilmlist,
     getGenres,
-    GenreInterface
+    GenreInterface,
+    getSearchResult
 } from "../../../services/filmService"
 import uniqid from 'uniqid';
 import './filmList.scss'
+import spinner from '/public/assets/spinner.gif'
+import { changeLoading } from "../../../redux/loading/action"
 
-const FilmsList = ({ sortType }: { sortType: number }) => {
+const FilmsList = () => {
 
     const [filmArr, setFilmArr] = useState<FilmInterface[]>([])
     const [genres, setGenres] = useState<GenreInterface[]>([])
 
     const dispatch = useDispatch()
-    const pageCounter = useSelector((state: any) => state.pageCounter.count)
-    const categories = useSelector((state: any) => state.pageCounter.categories)
+    const pageCounter = useSelector((state: ActionTypeInterface) => state.pageCounter.count)
+    const categories = useSelector((state: ActionTypeInterface) => state.pageCounter.categories)
+    const sortType = useSelector((state: ActionTypeInterface) => state.sortType.type)
+    const isLoaded = useSelector((state: ActionTypeInterface) => state.movieLoader.isLoaded)
+
+    const search = useSelector((state: ActionTypeInterface) => state.mainBannerAndSearch.search)
 
     useEffect(() => {
-        if (pageCounter > 1) {
-            getFilmlist(categories, pageCounter)
-                .then(res => setFilmArr(() => ([...filmArr, ...res.results])))
+        if (search.length > 0) {
+            // const handle: ReturnType<typeof setTimeout> = setTimeout(() => {
+            getSearchResult(search)
+                .then(res => {
+                    if (res.results.length > 0) {
+                        setFilmArr(res.results)
+                        dispatch(changeLoading(true))
+                    } else {
+                        console.log('e')
+                    }
+
+                })
+            // }, 1000)
+
+            // return () => clearTimeout(handle)
         } else {
-            getFilmlist(categories, pageCounter)
-                .then(res => setFilmArr(res.results))
+            if (pageCounter > 1) {
+                getFilmlist(categories, pageCounter)
+                    .then(res => {
+                        setFilmArr(() => ([...filmArr, ...res.results]))
+                        dispatch(changeLoading(true))
+                    })
+            } else {
+                getFilmlist(categories, pageCounter)
+                    .then(res => {
+                        setFilmArr(res.results)
+                        dispatch(changeLoading(true))
+                    })
+            }

```

Comment:
you can make the check like this, which is much shorter 

                getFilmlist(categories, pageCounter)
                    .then(res => {
                      pageCounter >1 ?  setFilmArr(() => ([...filmArr, ...res.results])) : setFilmArr(res.results)
                        dispatch(changeLoading(true))
                    })